compiler/rustc_hir_analysis/src/check/method/suggest.rs

```diff
@@ -1201,6 +1203,71 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
         false
     }
 
+    fn suggest_range_for_iter(
+        &self,
+        tcx: TyCtxt<'tcx>,
+        actual: Ty<'tcx>,
+        source: SelfSource<'tcx>,
+        span: Span,
+        item_name: Ident,
+        ty_str: &str,
+    ) -> bool {
+        if let SelfSource::MethodCall(expr) = source {
+            for (_, parent) in tcx.hir().parent_iter(expr.hir_id) {
```

Comment:
Is using `parent_iter` here necessary? Why do we care about more than one parent? Or should we only skip method call parents?

Reply:
For the case like this code:
```rust
let _res = a..b.take(1).sum();
```

`expr` is with span `b`, 
then expand parent to `b.take(1)`,
then expand parent to `b.take(1).sum()`,
then expand parent to `a..b.take(1).sum()`, this is our wanted expr.

Maybe we should add a search limit for the loop, for example 5?
since the possibility is less and less with the search number increasing.